### ISSUE TITLE
rook cluster fails after restart k8s node

### ISSUE THREAD

**Issue by beatlejuse** [rook/rook#1278]
i'm created operator, cluster and filesystem:
```
              apiVersion: rook.io/v1alpha1
              kind: Filesystem
              metadata:
                name: rookfs
                namespace: rook-cluster
              spec:
                metadataPool:
                  replicated:
                    size: 3
                dataPools:
                  - erasureCoded:
                     codingChunks: 1
                     dataChunks: 2
                metadataServer:
                  activeCount: 1
                  activeStandby: true
```
and try to connect to it:
```
apiVersion: v1
kind: Pod
metadata:
  name: ceph-tools
  namespace: rook-cluster
spec:
  containers:
  - name: ceph-tools
    image: nginx:1.13.5-alpine
    imagePullPolicy: IfNotPresent
    volumeMounts:
      - name: ceph-store
        mountPath: /srv/ceph
  volumes:
    - name: ceph-store
      flexVolume:
        driver: rook.io/rook
        fsType: ceph
        options:
          fsName: rookfs
          clusterName: rook-cluster
          path: /
```
and have such error:
```
  12m   1m      6       kubelet, 10.1.29.25             Warning FailedMount     Unable to mount volumes for pod "ceph-tools_rook-cluster(6ac879d0-d672-11e7-88d3-0050569d5b15)": timeout expired waiting for volumes to attach/mount for pod "rook-cluster"/"ceph-tools". list of unattached/unmounted volumes=[ceph-store]
  12m   1m      6       kubelet, 10.1.29.25             Warning FailedSync      Error syncing pod
  9m    13s     2       kubelet, 10.1.29.25             Warning FailedMount     MountVolume.SetUp failed for volume "ceph-store" : mount command failed, status: Failure, reason: failed to mount filesystem rookfs to /var/lib/kubelet/pods/6ac879d0-d672-11e7-88d3-0050569d5b15/volumes/rook.io~rook/ceph-store with monitor 10.3.183.65:6790,10.3.65.12:6790,10.3.131.61:6790:/ and options [name=admin secret=AQBNCiFa0oqjKhAA2WH+8VwHB0y17Irg6HCmIw== mds_namespace=rookfs]: mount failed: exit status 32
Mounting command: mount
Mounting arguments: 10.3.183.65:6790,10.3.65.12:6790,10.3.131.61:6790:/ /var/lib/kubelet/pods/6ac879d0-d672-11e7-88d3-0050569d5b15/volumes/rook.io~rook/ceph-store ceph [name=admin secret=AQBNCiFa0oqjKhAA2WH+8VwHB0y17Irg6HCmIw== mds_namespace=rookfs]
Output: mount: mount 10.3.183.65:6790,10.3.65.12:6790,10.3.131.61:6790:/ on /var/lib/kubelet/pods/6ac879d0-d672-11e7-88d3-0050569d5b15/volumes/rook.io~rook/ceph-store failed: Connection timed out
```

**Comment by beatlejuse** [comment]
in logs i see:
`with monitor 10.3.183.65:6790,10.3.65.12:6790,10.3.131.61:6790:/ `
but in "kubectl get svc" i see:
```
rook-cluster   rook-ceph-mon0         10.3.83.176    <none>        6790/TCP                1h
rook-cluster   rook-ceph-mon1         10.3.159.125   <none>        6790/TCP                1h
rook-cluster   rook-ceph-mon2         10.3.2.222     <none>        6790/TCP                1h
rook-cluster   rook-ceph-mon8         10.3.65.12     <none>        6790/TCP                1h
```
I did reboot some of the nodes in the cluster, resulting in a changed monitors. and now the file system connect does not work

**Comment by kokhang** [comment]
@beatlejuse the filesystem mount logic gets its monitoring endpoint from the configmap.
If you still have this environment, can you run `kubectl -n rook get configmaps rook-ceph-mon-endpoints -o yaml`?

Are the endpoints in the configmaps correct or stale? Rook has logic to update those when a new monitor gets added.

**Comment by beatlejuse** [comment]
Yes, here it is:
```
kubectl -n rook-cluster get configmaps rook-ceph-mon-endpoints -o yaml
apiVersion: v1
data:
  data: rook-ceph-mon0=10.3.118.3:6790,rook-ceph-mon1=10.3.7.253:6790,rook-ceph-mon3=10.3.219.65:6790
  mapping: '{"node":{"rook-ceph-mon0":{"Name":"10.1.29.21","Address":"10.1.29.21"},"rook-ceph-mon1":{"Name":"10.1.29.22","Address":"10.1.29.22"},"rook-ceph-mon3":{"Name":"10.1.29.25","Address":"10.1.29.25"}},"port":{}}'
  maxMonId: "3"
kind: ConfigMap
metadata:
  creationTimestamp: 2017-12-01T12:24:38Z
  name: rook-ceph-mon-endpoints
  namespace: rook-cluster
  resourceVersion: "662291"
  selfLink: /api/v1/namespaces/rook-cluster/configmaps/rook-ceph-mon-endpoints
  uid: 992f3a44-d692-11e7-bd76-0050569d694f
```
but listed here addresses - 10.1.29.21","10.1.29.22","10.1.29.25" - it's address of hardware cluster nodes.

**Comment by beatlejuse** [comment]
yes, configmap updated:
```
apiVersion: v1
data:
  data: rook-ceph-mon3=10.3.219.65:6790,rook-ceph-mon0=10.3.118.3:6790,rook-ceph-mon4=10.3.213.37:6790
  mapping: '{"node":{"rook-ceph-mon0":{"Name":"10.1.29.21","Address":"10.1.29.21"},"rook-ceph-mon3":{"Name":"10.1.29.25","Address":"10.1.29.25"},"rook-ceph-mon4":{"Name":"10.1.29.23","Address":"10.1.29.23"}},"port":{}}'
  maxMonId: "4"
kind: ConfigMap
metadata:
  creationTimestamp: 2017-12-01T12:24:38Z
  name: rook-ceph-mon-endpoints
  namespace: rook-cluster
  resourceVersion: "665473"
  selfLink: /api/v1/namespaces/rook-cluster/configmaps/rook-ceph-mon-endpoints
  uid: 992f3a44-d692-11e7-bd76-0050569d694f
```

**Comment by kokhang** [comment]
@beatlejuse Is the filesystem mount still trying to use old monitor endpoints even though the configmap has been updated?

**Comment by beatlejuse** [comment]
no, if i try to use the file system (when i deploy the pod, using it) on a physical node fell down Docker

**Comment by beatlejuse** [comment]
@kokhang @hunter @weekface @dclausen @phsiao 
I have a limited amount of time can test a connection error CEPH!
after a reboot one of the node cluster, the container with the Prometheus can not access the data. in the logs container no errors, but if i go inside the container and do "ls /data/mount" command just freeze.
I have the day to explore the issue, then I have to erase the cluster and rebuild it again because it need for work

**Comment by beatlejuse** [comment]
... and restart prometheus pod does not solve the problem

**Comment by galexrt** [comment]
@beatlejuse Could you please hop on our Slack so we can help you faster with the CephFS issue?
Addtionally could you enter one of the rook Pods (mon, OSD, any rook Pod would do) and run `rook version` and post that in the Slack then please :)

**Comment by beatlejuse** [comment]
@galexrt  yes, i'm in slack!

**Comment by kokhang** [comment]
@beatlejuse can you explain how you repo this? It seems you took a node which made the cephfs not able to connect to the cluster anymore. Is there more to it?

**Comment by beatlejuse** [comment]
@galexrt 
I tested today the cluster - the problem no longer occurs! ))

@kokhang 
I have kubernet cluster with 3 master nodes and one minion. on top of it deployed rook and monitoring. I tested the fault tolerance of multi-master cluster schema and selectively reboot one node. after the first reboot the node on which Prometheus was connected to the file system ceph, pod Prometheus was unable to start because it could not connect to the ceph.

**Comment by beatlejuse** [comment]
the problem came back again.
i'm restarting one node and after this, service can't connect to ceph.
`kubectl get po --all-namespaces -o wide
NAMESPACE      NAME                                    READY     STATUS              RESTARTS   AGE       IP           NODE
monitoring     prometheus-54968887c9-k9fjm             0/1       ContainerCreating   0          5m        <none>       10.1.29.22
rook-cluster   rook-api-847487f4f7-mhngf               1/1       Running             0          4d        10.4.102.3   10.1.29.25
rook-cluster   rook-ceph-mds-rookfs-7bc8dc5d-8zmrw     1/1       Unknown             0          4d        10.4.16.8    10.1.29.21
rook-cluster   rook-ceph-mds-rookfs-7bc8dc5d-jvbk7     1/1       Running             0          5m        10.4.76.7    10.1.29.22
rook-cluster   rook-ceph-mds-rookfs-7bc8dc5d-qzw9q     1/1       Running             0          4d        10.4.102.6   10.1.29.25
rook-cluster   rook-ceph-mgr0-99f65c45b-rb7pd          1/1       Running             0          4d        10.4.102.4   10.1.29.25
rook-cluster   rook-ceph-mgr1-594b89dc46-8zs8v         1/1       Running             0          4d        10.4.102.5   10.1.29.25
rook-cluster   rook-ceph-mon0-5xcpf                    1/1       Unknown             0          4d        10.4.16.4    10.1.29.21
rook-cluster   rook-ceph-mon3-qj6dg                    1/1       Running             2          4d        10.4.102.8   10.1.29.25
rook-cluster   rook-ceph-mon4-gnr6c                    1/1       Running             4          2d        10.4.34.4    10.1.29.23
rook-cluster   rook-ceph-mon5-h9bl9                    1/1       Running             0          27s       10.4.76.9    10.1.29.22
rook-cluster   rook-ceph-osd-4pjxb                     1/1       Running             0          2d        10.4.76.2    10.1.29.22
rook-cluster   rook-ceph-osd-l84rj                     1/1       Running             0          3d        10.4.34.2    10.1.29.23
rook-cluster   rook-ceph-osd-pggnv                     1/1       Running             0          4d        10.4.102.2   10.1.29.25
rook-cluster   rook-ceph-osd-q4zgn                     1/1       NodeLost            0          4d        10.4.16.5    10.1.29.21
rook-cluster   rook-tools-558fc7867d-p7c76             1/1       Running             0          2d        10.4.34.5    10.1.29.23
rook-system    rook-agent-9lj8t                        1/1       Running             0          3d        10.1.29.23   10.1.29.23
rook-system    rook-agent-lggtj                        1/1       Running             0          4d        10.1.29.25   10.1.29.25
rook-system    rook-agent-mhdl2                        1/1       Running             0          2d        10.1.29.22   10.1.29.22
rook-system    rook-agent-qmh44                        1/1       NodeLost            0          4d        10.1.29.21   10.1.29.21
rook-system    rook-operator-54dbd6d85d-2b5zf          1/1       Unknown             0          4d        10.4.16.3    10.1.29.21
rook-system    rook-operator-54dbd6d85d-5v5vh          1/1       Running             0          5m        10.4.76.8    10.1.29.22`

`

 kubectl -n monitoring describe po prometheus-54968887c9-k9fjm
Name:           prometheus-54968887c9-k9fjm
Namespace:      monitoring
Node:           10.1.29.22/10.1.29.22
Start Time:     Mon, 18 Dec 2017 10:05:58 +0300
Labels:         app=prometheus
                component=core
                pod-template-hash=1052444375
Annotations:    kubernetes.io/created-by={"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicaSet","namespace":"monitoring","name":"prometheus-54968887c9","uid":"22ec2675-e018-11e7-a45e-0050569...
Status:         Pending
IP:
Controllers:    ReplicaSet/prometheus-54968887c9
Containers:
  prometheus:
    Container ID:
    Image:              quay.io/prometheus/prometheus:v2.0.0
    Image ID:
    Port:               9090/TCP
    Args:
      --config.file=/etc/prometheus/prometheus.yml
      --storage.tsdb.path=/prometheus
      --web.console.libraries=/usr/share/prometheus/console_libraries
      --web.console.templates=/usr/share/prometheus/consoles
    State:              Waiting
      Reason:           ContainerCreating
    Ready:              False
    Restart Count:      0
    Limits:
      cpu:      500m
      memory:   2500Mi
    Requests:
      cpu:      100m
      memory:   100Mi
    Environment:
      STORAGE_RETENTION:        <set to the key 'storage-retention' of config map 'prometheus-env'>     Optional: false
      STORAGE_MEMORY_CHUNKS:    <set to the key 'storage-memory-chunks' of config map 'prometheus-env'> Optional: false
      EXTERNAL_URL:             <set to the key 'external-url' of config map 'prometheus-env'>          Optional: false
    Mounts:
      /etc/prometheus from config-volume (rw)
      /etc/prometheus-rules from rules-volume (rw)
      /prometheus from prometheus-data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-jn44d (ro)
Conditions:
  Type          Status
  Initialized   True
  Ready         False
  PodScheduled  True
Volumes:
  prometheus-data:
  <unknown>
  config-volume:
    Type:       ConfigMap (a volume populated by a ConfigMap)
    Name:       prometheus-configmap
    Optional:   false
  rules-volume:
    Type:       ConfigMap (a volume populated by a ConfigMap)
    Name:       prometheus-rules
    Optional:   false
  default-token-jn44d:
    Type:       Secret (a volume populated by a Secret)
    SecretName: default-token-jn44d
    Optional:   false
QoS Class:      Burstable
Node-Selectors: master=true
Tolerations:    <none>
Events:
  FirstSeen     LastSeen        Count   From                    SubObjectPath   Type            Reason                  Message
  ---------     --------        -----   ----                    -------------   --------        ------                  -------
  6m            6m              1       default-scheduler                       Normal          Scheduled               Successfully assigned prometheus-54968887c9-k9fjm to 10.1.29.22
  6m            6m              1       kubelet, 10.1.29.22                     Normal          SuccessfulMountVolume   MountVolume.SetUp succeeded for volume "config-volume"
  6m            6m              1       kubelet, 10.1.29.22                     Normal          SuccessfulMountVolume   MountVolume.SetUp succeeded for volume "rules-volume"
  6m            6m              1       kubelet, 10.1.29.22                     Normal          SuccessfulMountVolume   MountVolume.SetUp succeeded for volume "default-token-jn44d"
  3m            3m              1       kubelet, 10.1.29.22                     Warning         FailedMount             MountVolume.SetUp failed for volume "prometheus-data" : mount command failed, status: Failure, reason: failed to mount filesystem rookfs to /var/lib/kubelet/pods/e60a61dc-e3c1-11e7-922b-0050569d694f/volumes/rook.io~rook/prometheus-data with monitor 10.3.10.38:6790,10.3.61.57:6790,10.3.249.159:6790:/webdev/prometheus and options [name=admin secret=AQAZQTFatGnCDxAAkBvLdUQCG6uKUeNhKLGqhQ== mds_namespace=rookfs]: mount failed: exit status 32
Mounting command: mount
Mounting arguments: 10.3.10.38:6790,10.3.61.57:6790,10.3.249.159:6790:/webdev/prometheus /var/lib/kubelet/pods/e60a61dc-e3c1-11e7-922b-0050569d694f/volumes/rook.io~rook/prometheus-data ceph [name=admin secret=AQAZQTFatGnCDxAAkBvLdUQCG6uKUeNhKLGqhQ== mds_namespace=rookfs]
Output: mount: 10.3.10.38:6790,10.3.61.57:6790,10.3.249.159:6790:/webdev/prometheus: can't read superblock


  4m    2m      2       kubelet, 10.1.29.22             Warning FailedMount     Unable to mount volumes for pod "prometheus-54968887c9-k9fjm_monitoring(e60a61dc-e3c1-11e7-922b-0050569d694f)": timeout expired waiting for volumes to attach/mount for pod "monitoring"/"prometheus-54968887c9-k9fjm". list of unattached/unmounted volumes=[prometheus-data]
  4m    2m      2       kubelet, 10.1.29.22             Warning FailedSync      Error syncing pod
  11s   11s     1       kubelet, 10.1.29.22             Warning FailedMount     MountVolume.SetUp failed for volume "prometheus-data" : mount command failed, status: Failure, reason: failed to mount filesystem rookfs to /var/lib/kubelet/pods/e60a61dc-e3c1-11e7-922b-0050569d694f/volumes/rook.io~rook/prometheus-data with monitor 10.3.249.159:6790,10.3.10.38:6790,10.3.61.57:6790:/webdev/prometheus and options [name=admin secret=AQAZQTFatGnCDxAAkBvLdUQCG6uKUeNhKLGqhQ== mds_namespace=rookfs]: mount failed: exit status 32
Mounting command: mount
Mounting arguments: 10.3.249.159:6790,10.3.10.38:6790,10.3.61.57:6790:/webdev/prometheus /var/lib/kubelet/pods/e60a61dc-e3c1-11e7-922b-0050569d694f/volumes/rook.io~rook/prometheus-data ceph [name=admin secret=AQAZQTFatGnCDxAAkBvLdUQCG6uKUeNhKLGqhQ== mds_namespace=rookfs]
Output: mount: 10.3.249.159:6790,10.3.10.38:6790,10.3.61.57:6790:/webdev/prometheus: can't read superblock`

`kubectl -n rook-cluster exec -it rook-tools-558fc7867d-p7c76 bash
root@rook-tools-558fc7867d-p7c76:/# ceph -s
  cluster:
    id:     f1d80266-8b69-4516-bdf1-9fd73c2ccb21
    health: HEALTH_WARN
            Degraded data redundancy: 526/1080 objects degraded (48.704%), 200 pgs unclean, 200 pgs degraded, 200 pgs undersized
            18 slow requests are blocked > 32 sec

  services:
    mon: 3 daemons, quorum rook-ceph-mon4,rook-ceph-mon3,rook-ceph-mon5
    mgr: rook-ceph-mgr1(active), standbys: rook-ceph-mgr0
    mds: rookfs-1/1/1 up  {0=mqzw9q=up:active}, 1 up:standby-replay
    osd: 4 osds: 2 up, 2 in

  data:
    pools:   2 pools, 200 pgs
    objects: 360 objects, 1207 MB
    usage:   5333 MB used, 51513 MB / 56846 MB avail
    pgs:     526/1080 objects degraded (48.704%)
             200 active+undersized+degraded
`

**Comment by beatlejuse** [comment]
it's real big problem.
today in another k8s cluster appear this problem WITHOUT any changes, after 4 days of clucter was build.
i'm rebuild cluster and this problem appear immediateli after cluster was build and some services was deploying into it.
when i'm deploing another service, it's can't connect to ceph fs.
after this, any services can't connect to ceph:
https://gist.github.com/beatlejuse/631c00b59cc2f1cfafb3e57fc16e3724

**Comment by beatlejuse** [comment]
logs1:
https://gist.github.com/beatlejuse/93b13a12a33e84b2d901faa7db8fabb5
logs2:
https://gist.github.com/beatlejuse/262517d18666b81df16f469343355320
ceph osd tree:
https://gist.github.com/beatlejuse/737a2119f197baa82ec0f60b4de929a1

**Comment by kokhang** [comment]
Hi @beatlejuse one thing that i noticed from the log is that you are running Rook v0.6.0-38.gdf22444. This build does not have a fix for an issue that causes the rook-api to use 100% cpu. This causes the whole node to be degraded and a lot of other issues could come from it. This is the bug: https://github.com/rook/rook/issues/1195. I suggest you to use Rook v0.6.1 at a minimum.

From the logs i see that the monitoring are failing and Rook cannot connect to it. The rook-agent mounts the cephfs by connecting to the monitor endpoint. When this issue happens, ensure that the monitoring endpoint that is connecting to (logged in the rook-agent) matches the monitor endpoints in the monitor endpoint configmap and also matches with the monitor svc endpoints.

Another issue you to help debug is to use the rookctl in the rook-toolbox and run `rookctl filesystem mount --name myfs --path /tmp/registry`. This will tell whether the issue is with the monitor connectivity or something else.

But first, i would suggest to try Rook 0.6.1+. And see if the issues you are running with the monitor not working is due to the CPU usage bug by the rook-api.

**Comment by beatlejuse** [comment]
I upgraded to 0.6.2.
CPU loading API has decreased significantly.
reset successively all three master nodes.
the problem with the connection to the file system are not observed.
I will try to conduct all the tests again in a few days.

**Comment by beatlejuse** [comment]
after testing reboot the node, there is a problem with the mons:
ceph -s
https://gist.github.com/beatlejuse/089507aee7c56318b9ef3e8206ba3831
ceph osd tree
https://gist.github.com/beatlejuse/b6d5534d9313802ccc29e6100dae2b6a
logs1
http://www.filehosting.org/file/details/714825/rook-logs.gz
logs2
https://gist.github.com/beatlejuse/18070675f9833c4d06790f3a32bf1cba

**Comment by beatlejuse** [comment]
how can I fix this until we fix the automatic rebuilding of OSD?

**Comment by beatlejuse** [comment]
today I re-tested the reboot a node in the cluster.
the connection to the file system СЕРН is still not working.
https://gist.github.com/beatlejuse/fb7a225317d6d225a2aa9a1ccb920d7a
https://gist.github.com/beatlejuse/1aef1995955ed9e99db6604b7c7c6882
kubectl get svc --all-namespaces
https://gist.github.com/beatlejuse/0daa528cf0ddc4639bbc7e559e9237ed
ceph -s
https://gist.github.com/beatlejuse/a6305cd68dd4e7853321db7d76669b24
logs1
http://www.filehosting.org/file/details/715499/rook-logs
logs2
https://gist.github.com/beatlejuse/b75f693937960d7b15e4971ebea11262

**Comment by travisn** [comment]
To summarize the latest slack conversation...

Repro:
- Start the cluster (0.6.2)
- Verify the ceph health, all the mons are healthy and osds are `up` and `in`
- Create a filesystem
- Connect a pod to the filesystem
- Verify everything is healthy
- Start rebooting the nodes to verify resiliency. 
- Observe the mons will become unhealthy and failover, and some osds will be `out`

@beatlejuse could you provide more details on the rebooting step to repro? Are you restarting the nodes one at a time? Are the unhealthy issues immediate or after some amount of time? Any details you have would be helpful so we can repro on our side.

A couple of observations on the health issues:
1) The mons are failing over. Something is causing them to be down for at least 5 minutes and then the operator fails them over. After all the original mons failover, the clients of cephfs will likely fail as well until we have dynamic provisioning (https://github.com/rook/rook/issues/1125). 
2) The osds are failing, which causes the writes/reads to fail. It's not clear from the logs what is causing them to be out

**Comment by beatlejuse** [comment]
yes, i restarting the nodes one at a time. 
I'll try to make a video ))

**Comment by beatlejuse** [comment]
I made a video!
https://youtu.be/6wLk9Ey581o
legend.
in the beginning it is shown that in case of breakage ceph Prometheus cannot mount a folder to connect to ceph.
4:06 - stop all the node in the cluster, wipe their hdd and start.
next, the nodes need to download the CoreOS PXE image and the script of the matchbox set.
14:25 - deploy ceph and creation of the ceph-fs
16:27 - deploy ceph-tools (rook-tools with intergrown ssh and mounted ceph fs), and create folders in ceph-fs.
18:04 - listing folders in ceph-fs
24:12 - restart the first node
29:24 - rebooting the second node
48:22 - rebooting the third node
49:29 - you can see the problem when you restart the server, mon1 blocks the system and does not give CoreOS a normal reboot
50:50 - you can see the problem with connecting to ceph-fs - ceph-tools cannot read mounted folder
55:14 - you can see the problem of Prometheus connecting to ceph-fs
57:13 - ceph-fs unusable. remove the ceph cluster
52:26 - you can see the problem with libceph
1:01:02 - creating ceph cluster
1:03:44 ceph cluster has not been created, delete the operator
1:04:15 - you can see the problem with libceph - ceph completely removed, and libceph still gives error
1:05:50 - create a ceph cluster
1:06:54 - you can see the problem with creating a cluster - do not accept old data, in order to raise the ceph would have to re-delete all data - we irretrievably lose valuable data!
1:08:46 - creating ceph operator
1:08:56 - creating ceph cluster
1:09:57 - create folders in ceph-fs (it does the task in gitlab)
1:09:57 - connecting to ceph-tools and watch the contents of the ceph-fs
1:10:51 - restart Prometheus

**Comment by beatlejuse** [comment]
@kokhang  @travisn  @galexrt there is a chance to solve my problem?

**Comment by galexrt** [comment]
@beatlejuse I just noticed you are using `erasureCoded` for the `dataPools`. The issue with data deletion/overwrite is related to the operator not setting the overwrite "flag" for CephFS.
I suggest you try to replicate the issue with `replicated` for the `dataPools`.

Another point just as a recommendation is that for something like Prometheus, MySQL, Postgres, etc (applications that can't access their data in parallel/from different instances), it is better to use RBD as it should be faster for applications like that.
CephFS is for applications that can access their files in parallel/from multiple instances at the same time, examples: Minio (use Ceph Object storage as an alternative), GitLab, PHP web applications (NextCloud, etc).

**Comment by beatlejuse** [comment]
это странно. я вроде бы не использую erasureCoded.
конфиг файловой системы выглядит у меня так:
```
              apiVersion: rook.io/v1alpha1
              kind: Filesystem
              metadata:
                name: rookfs
                namespace: rook-cluster
              spec:
                metadataPool:
                  replicated:
                    size: 2
                dataPools:
                  - replicated:
                      size: 2
                metadataServer:
                  activeCount: 1
                  activeStandby: true
```
на какой минуте видно erasureCoded?

**Comment by galexrt** [comment]
@beatlejuse You have posted your manifests in your first comment (see https://github.com/rook/rook/issues/1278#issue-278397981) and in that `erasureCoded` is used in `dataPools`.

**Comment by beatlejuse** [comment]
@galexrt it was a very long time. I often change the configs trying to configure the system

**Comment by kokhang** [comment]
@beatlejuse do you have a set of steps to reproduce this issue and get it to this state? Or does it become apparent after a while of usage? Your ceph cluster doesnt look good in https://github.com/rook/rook/issues/1278#issuecomment-353848520.

**Comment by beatlejuse** [comment]
@kokhang  of course there is!
I even made the video all the steps: https://github.com/rook/rook/issues/1278#issuecomment-356918438

**Comment by beatlejuse** [comment]
@kokhang also you can shutdown k8s cluster (all nodes) and start it again.
you will see many interesting errors))

**Comment by beatlejuse** [comment]
@kokhang  @galexrt 
Any idea about this problem?
I recall, when I restart the node k8s-cluster, after rebooting lost mons and is not automatically restored.
Is this problem just me?
when you restart the node k8s-cluster,  ceph-cluster is not destroyed?
If so, could you share for me the full set of operating YAML configs to operator and cluster?
I now use the image rook/rook:v0.6.2 .
as far as I can see, you no longer develop v0.6.2 .
instead, you develop version v0.6.0 .
What is the reason? maybe I should use the version v0.6.0 ?

**Comment by beatlejuse** [comment]
@kokhang @galexrt ping

**Comment by beatlejuse** [comment]
during the development of this issue there have been many changes. I will sum up:
1. i have k8s v1.8.1 on CoreOS 1520.8.0 on bare-metal (vmware).
2. I renamed back the namespace "rook-cluster" to "rook"
3. I rolled back from version 0.6.2 to version 0.6.0.215

I again rebooted one node kubernetes cluster and see the following situation:
```rook          rook-api-57d695d7dc-ns6fj               1/1       Running   0          16h       10.1.29.25   10.1.29.25
rook          rook-ceph-mds-rookfs-6cf6d774bd-rcs7g   1/1       Running   0          16h       10.1.29.22   10.1.29.22
rook          rook-ceph-mds-rookfs-6cf6d774bd-z96qj   1/1       Running   0          16h       10.1.29.23   10.1.29.23
rook          rook-ceph-mgr0-76964b8fd6-bzm7w         1/1       Running   1          16h       10.1.29.23   10.1.29.23
rook          rook-ceph-mon0-69276                    1/1       Running   0          16h       10.1.29.21   10.1.29.21
rook          rook-ceph-mon1-vwgd7                    1/1       Running   0          16h       10.1.29.22   10.1.29.22
rook          rook-ceph-mon2-j46bj                    1/1       Running   1          16h       10.1.29.23   10.1.29.23
rook          rook-ceph-osd-6dtbk                     1/1       Running   0          16h       10.1.29.21   10.1.29.21
rook          rook-ceph-osd-b5wcp                     1/1       Running   0          16h       10.1.29.24   10.1.29.24
rook          rook-ceph-osd-gz55f                     1/1       Running   1          16h       10.1.29.23   10.1.29.23
rook          rook-ceph-osd-plwnv                     1/1       Running   1          16h       10.1.29.25   10.1.29.25
rook          rook-ceph-osd-sw6zq                     1/1       Running   0          16h       10.1.29.22   10.1.29.22
rook          rook-tools-576f7bf7df-vz5t4             1/1       Running   0          16h       10.4.3.2     10.1.29.25
rook-system   rook-agent-8n69w                        1/1       Running   0          16h       10.1.29.22   10.1.29.22
rook-system   rook-agent-gkbf2                        1/1       Running   0          16h       10.1.29.24   10.1.29.24
rook-system   rook-agent-lxhl7                        1/1       Running   0          16h       10.1.29.21   10.1.29.21
rook-system   rook-agent-nvqrx                        1/1       Running   0          16h       10.1.29.23   10.1.29.23
rook-system   rook-agent-srzg2                        1/1       Running   0          16h       10.1.29.25   10.1.29.25
```

```ceph osd tree
ID  CLASS WEIGHT  TYPE NAME           STATUS REWEIGHT PRI-AFF
 -1       0.10437 root default
 -9       0.02609     host 10-1-29-21
  2   hdd 0.02609         osd.2           up  1.00000 1.00000
 -3       0.02609     host 10-1-29-22
  0   hdd 0.02609         osd.0           up  1.00000 1.00000
 -4       0.02609     host 10-1-29-23
  3   hdd 0.02609         osd.3           up  1.00000 1.00000
 -2             0     host 10-1-29-24
-11       0.02609     host 10-1-29-25
  4   hdd 0.02609         osd.4           up  1.00000 1.00000
```
```ceph status
  cluster:
    id:     44edeb8a-1800-46ae-bf1d-61996e59d3e1
    health: HEALTH_OK

  services:
    mon: 3 daemons, quorum rook-ceph-mon0,rook-ceph-mon1,rook-ceph-mon2
    mgr: rook-ceph-mgr0(active)
    mds: rookfs-1/1/1 up  {0=mrcs7g=up:active}, 1 up:standby-replay
    osd: 4 osds: 4 up, 4 in

  data:
    pools:   2 pools, 200 pgs
    objects: 21 objects, 2246 bytes
    usage:   15380 MB used, 94216 MB / 107 GB avail
    pgs:     200 active+clean

  io:
    client:   1279 B/s rd, 2 op/s rd, 0 op/s wr
```

**Comment by beatlejuse** [comment]
and this is my deploy configs:
https://gist.github.com/beatlejuse/9ca3156dacaf0abd391b0d125326b5b8
https://gist.github.com/beatlejuse/913eb1a6b23b1199d23c83eec207c2f4

**Comment by travisn** [comment]
@beatlejuse  ok so the issue is that osd.1 isn't coming back? I see the osd pod running, but the osd is not found in ceph.
@jbw976 any ideas for why osd.1 would be in the overall [osd dump](https://github.com/rook/rook/blob/master/pkg/daemon/ceph/osd/health.go#L63), but then not be found when [looked up by id](https://github.com/rook/rook/blob/master/pkg/daemon/ceph/client/osd.go#L94)?

**Comment by jbw976** [comment]
This looks like #1477 that was fixed in master just yesterday.  Starting with `rook/rook:v0.6.0-219.g09dc9b8`, this should not happen anymore.  Once getting the rook container image with the fix, you may need to mark the OSD as back in the cluster and give it a weight similar to other OSDs in the cluster that are the same size.
```
ceph osd crush reweight osd.1 <weight value>
ceph osd in 1
```

**Comment by beatlejuse** [comment]
@travisn  @jbw976  ok!
i'm rebuild my k8s cluster and update rook image to 0.6.0.219
after restart one node i see this:
ceph status 
https://gist.github.com/beatlejuse/9a5afa8ef80d849d0227a906d1a75177
ceph osd tree
https://gist.github.com/beatlejuse/687e0611446dcd19f2de807f9d3927fb
>> ceph osd crush reweight osd.0 5
ceph osd tree
https://gist.github.com/beatlejuse/b281a69a2968c34e721cf3fe102f207a
>>  ceph osd in 0
ceph osd tree
https://gist.github.com/beatlejuse/3c20d42c8d67efc6995e5f5529e4b428

I will summarize the experiment:
1. when I restart one node, the OSD does not automatically recover, as claimed galexrt.
2. command "ceph osd crush reweight osd.1 <weight value>" and "ceph osd in 1" not enough to return OSD back.

**Comment by beatlejuse** [comment]
@travisn @jbw976  I found the cause of my problem!
I have a var/lib/docker cleanup step in my node load script to remove old containers and images.
if I disable this step, the node returns to the ceph cluster when it reboots.
as I understand it, when cleaning var/lib/docker node is necessary to add not as existing, and as new.
how can I do that?

**Comment by travisn** [comment]
@beatlejuse not sure I follow, what are you trying to accomplish by deleting /var/lib/docker? If you're trying to reboot the node and have the osd recover after restart, you would need to keep the docker folder.

**Comment by beatlejuse** [comment]
@travisn 
the folder /var/lib/docker and exactly /var/lib/docker/overlay consumes too much space.
if the disk runs out of free space, the reason for this is the /var/lib/docker/overlay folder.
as you know, kubernetes exclude the node from the cluster, if free space becomes less than 20%.
clearing the /var/lib/docker folder with the startup script allows the node to be returned to the cluster again with a simple reboot.

does rook have the ability to add new nodes to an existing cluster?
maybe it's easier for me to remove the reloaded node from the cluster and add it back as a new node?

**Comment by travisn** [comment]
Yes, in 0.7 Rook will add new nodes to the cluster. If you set `useAllNodes: true`, the new nodes should be picked up automatically. If you have `useAllNodes: false` and specify nodes individually, you would need to add the new node to the cluster crd.

**Comment by beatlejuse** [comment]
i using "useAllNodes: true".
I mean, I want the rook cluster to forget the failed node and add it to the cluster as a new one after it's back to normal. how can i tell rook cluster to forget the failed node?

**Comment by travisn** [comment]
From Rook's perspective, deleting the `dataDirHostPath` on the node would cause new OSDs to be created. The old OSDs would be marked as `down` and `out` by ceph.

**Comment by beatlejuse** [comment]
however, experience has shown that a new OSD is not created.
and me seems, that the old need to remove.
I think there should be an appropriate command.

**Comment by travisn** [comment]
you have deleted the `dataDirHostPath` as part of the reboot and it didn't help? I don't see how the old osds could stick around. What is the `state` of the osds in the `ceph osd status`?

**Comment by beatlejuse** [comment]
yes. https://github.com/rook/rook/issues/1278#issuecomment-366970355
but about this question we can until forget
---
now I'm testing a node reboot and found a new problem - after a consecutive reboot of all 5 nodes I can no longer connect to either ceph-fs or ceph-block
https://gist.github.com/beatlejuse/d6fc62e824772d2cc075b25ee56ad9ed
https://gist.github.com/beatlejuse/536baeb85d2de5f88e846ebb6651bd35
https://gist.github.com/beatlejuse/a681c044110d8ddfb34ed6b593c34abe
